source/by-role/system-admin/upgrade/upgrading-search.rst

```diff
@@ -0,0 +1,170 @@
+:is-up-to-date: False
+:last-updated: 4.1.0
+
+.. _upgrading-search:
+
+================
+Upgrading Search
+================
+
+Starting version 4.1.0, CrafterCMS now uses OpenSearch.
+This section describes how to upgrade search for CrafterCMS installed on a server.
+
+Please read through the upgrade instructions :ref:`here <upgrading-craftercms-on-a-server>` first.
+The steps for upgrading to OpenSearch follows almost exactly the same steps as listed in that document.
+
+--------------------------------------
+Upgrading 3.1.x -> 4.1.0 (from ES 6.x)
+--------------------------------------
+To upgrade your 3.1.x installation, we'll be running the upgrade scripts from a new binary archive.
+We'll use the ``upgrade-search.sh`` script, which will update the data in place. Please backup your data directory before running the script.
+
+Here is the ``upgrade-search`` script params:
+
+.. code-block:: text
+
+     -h,--help                                  Show usage information
+        --port <port>                           Elasticsearch port to use for
+                                                upgrade ES temporary instance
+        --status-retries <max status retries>   How many times to try to get a
+                                                yellow status from the ES
+                                                cluster (waiting 10s between
+                                                retries)
+        --status-timeout <seconds>              Timeout in seconds for the
+                                                status check of the ES cluster
+        --stay-alive                            Set to true to keep the
+                                                process alive after reindexing
+                                                is complete. This allows to
+                                                query the ES server and
+                                                review.
+    e.g:
+
+    # Run in a different port
+    ./upgrade-search.sh --port 9206 /path/of/install/to/be/upgraded --stay-aliveUpgrading 3.1.x -> 4.1.0 (from ES 6.x)
+
+Before beginning your upgrade, remember that the install you're upgrading needs to be running. Here are the steps to  upgrade your CrafterCMS  3.1.x install:
```

Comment:
@alhambrav Studio/Delivery needs to be **not** running before starting the upgrade process.